src/timezone/timezone.service.ts

```diff
@@ -0,0 +1,48 @@
+import * as moment from 'moment-timezone';
+import { Injectable } from '@nestjs/common';
+import { UserService } from '../modules/user/user.service';
+import { ITimezone } from '../common/interfaces/timezone.interface';
+import { majorTimezones } from '../common/utils/time-utils'; 
+
+@Injectable()
+export class TimezoneService {
+  constructor(private readonly userService: UserService) {}
+
+  getAllTimezones(): ITimezone[] {
+    return moment.tz.names().map(name => ({
+      id: name,
+      displayName: `${name} (${moment.tz(name).format('Z z')})`
+    }));
+  }
+
+  countAllTimezones(): number {
+    return moment.tz.names().length;
+  }
+
+  getMajorTimezones(): ITimezone[] {
+    return majorTimezones
+      .filter(timezone => moment.tz.names().includes(timezone))
+      .map(timezone => ({
+        id: timezone,
+        displayName: `${timezone} (${moment.tz(timezone).format('Z z')})`
+      }));
+  }
+
+  isValidTimezone(timezone: string): boolean {
+    return moment.tz.names().includes(timezone);
+  }
+
+  async updateUserTimezone(userId: string, timezone: string): Promise<any> {
```

Comment:
consider update user is a private service, it needs to validate user's authentication, then allow user to update timezone